Lib/test/test_socket.py

```diff
@@ -7327,19 +7327,32 @@ def test_dual_stack_client_v6(self):
 @requireAttrs(socket, "recv_fds")
 @requireAttrs(socket, "AF_UNIX")
 class SendRecvFdsTests(unittest.TestCase):
-    def testSendAndRecvFds(self):
-        def close_pipes(pipes):
-            for fd1, fd2 in pipes:
-                os.close(fd1)
-                os.close(fd2)
-
+    def _cleanup_fds(self, fds):
         def close_fds(fds):
             for fd in fds:
                 os.close(fd)
+        self.addCleanup(close_fds, fds)
+
+    def _test_pipe(self, rfd, wfd, msg):
+        assert len(msg) < 512
+        os.write(wfd, msg)
+        data = os.read(rfd, 512)
+        self.assertEqual(data, msg)
```

Comment:
```suggestion
        os.write(wfd, msg)
        data = os.read(rfd, len(msg))
        self.assertEqual(data, msg)
```

I think we can do it like this right?

Reply:
`man 7 pipe` says:

> POSIX.1 says that writes of less than PIPE_BUF bytes must be atomic: the output data is written to the pipe as a contiguous sequence. [...] POSIX.1 requires PIPE_BUF to be at least 512 bytes.

I hardcoded 512 here, which may look confusing. I've updated it to make it more clear.